[BUG] Prometheus metrics - panics when adding new policy

**Software version numbers**
State the version numbers of applications involved in the bug.

* Kyverno version: v1.4.0-rc1

**Describe the bug**
While I was applying a generate policy to the cluster, Kyverno panics with the following error message:
```
I0609 21:01:29.603821       1 validate_controller.go:261] PolicyController "msg"="policy created"  "kind"="ClusterPolicy" "name"="add-networkpolicy" "uid"="47c1f0b4-5a23-4462-af9b-8617983463cf"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1760631]

goroutine 1350932 [running]:
github.com/kyverno/kyverno/pkg/metrics/policyruleinfo.PromMetrics.AddPolicy(0xc00039c048, 0xc00039c050,0xc00039c058, 0xc00039c060, 0xc00039c068, 0x1c430e0, 0xc00530d380, 0xd, 0xc002f37a40)
        /kyverno/pkg/metrics/policyruleinfo/policyRuleInfo.go:53 +0x141
github.com/kyverno/kyverno/pkg/policy.(*PolicyController).registerPolicyRuleInfoMetricAddPolicy(0xc0001feb40, 0x253eee0, 0xc0005b6460, 0xc00530d380)
        /kyverno/pkg/policy/validate_controller.go:200 +0x9d
created by github.com/kyverno/kyverno/pkg/policy.(*PolicyController).addPolicy
        /kyverno/pkg/policy/validate_controller.go:264 +0x1e7
```

And it brought down other Kyverno Pods:
```
# k get pod -n kyverno
NAME                       READY   STATUS             RESTARTS   AGE
kyverno-6cf7477544-58vjq   1/1     Running            4          10h
kyverno-6cf7477544-85j7x   0/1     CrashLoopBackOff   3          37m
kyverno-6cf7477544-9pnz9   0/1     CrashLoopBackOff   4          10h
kyverno-6cf7477544-mp9tq   1/1     Running            6          10h
```

**To Reproduce**
I cannot reproduce it consistently, here's the policy I was using:
```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: add-networkpolicy
  annotations:
    policies.kyverno.io/category: Workload Management
    policies.kyverno.io/description: By default, Kubernetes allows communications across
      all pods within a cluster. Network policies and, a CNI that supports network policies,
      must be used to restrict communinications. A default NetworkPolicy should be configured
      for each namespace to default deny all ingress traffic to the pods in the namespace.
      Application teams can then configure additional NetworkPolicy resources to allow
      desired traffic to application pods from select sources.
spec:
  validationFailureAction: audit
  rules:
  - name: default-deny-ingress
    match:
      resources:
        kinds:
        - NetworkPolicy
        namespaces:
        - test-shuting
    generate:
      kind: NetworkPolicy
      name: "default-deny-ingress-{{request.object.metadata.uid}}"
      namespace: "{{request.object.metadata.namespace}}"
      synchronize : true
      data:
        spec:
          # select all pods in the namespace
          podSelector: {}
          policyTypes:
          - Ingress
```

--- a/pkg/metrics/parsers.go
+++ b/pkg/metrics/parsers.go
@@ -2,8 +2,9 @@ package metrics
 
 import (
 	"fmt"
-	kyverno "github.com/kyverno/kyverno/pkg/api/kyverno/v1"
 	"reflect"
+
+	kyverno "github.com/kyverno/kyverno/pkg/api/kyverno/v1"
 )
 
 func ParsePolicyValidationMode(validationFailureAction string) (PolicyValidationMode, error) {
@@ -17,10 +18,11 @@ func ParsePolicyValidationMode(validationFailureAction string) (PolicyValidation
 	}
 }
 
-func ParsePolicyBackgroundMode(backgroundMode bool) PolicyBackgroundMode {
-	if backgroundMode {
+func ParsePolicyBackgroundMode(backgroundMode *bool) PolicyBackgroundMode {
+	if backgroundMode == nil || *backgroundMode {
 		return BackgroundTrue
 	}
+
 	return BackgroundFalse
 }
 

--- a/pkg/metrics/policychanges/policyChanges.go
+++ b/pkg/metrics/policychanges/policyChanges.go
@@ -2,10 +2,11 @@ package policychanges
 
 import (
 	"fmt"
+	"time"
+
 	kyverno "github.com/kyverno/kyverno/pkg/api/kyverno/v1"
 	"github.com/kyverno/kyverno/pkg/metrics"
 	prom "github.com/prometheus/client_golang/prometheus"
-	"time"
 )
 
 func (pm PromMetrics) registerPolicyChangesMetric(
@@ -38,7 +39,7 @@ func (pm PromMetrics) RegisterPolicy(policy interface{}, policyChangeType Policy
 		if err != nil {
 			return err
 		}
-		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*inputPolicy.Spec.Background)
+		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(inputPolicy.Spec.Background)
 		policyType := metrics.Cluster
 		policyNamespace := "" // doesn't matter for cluster policy
 		policyName := inputPolicy.ObjectMeta.Name
@@ -51,7 +52,7 @@ func (pm PromMetrics) RegisterPolicy(policy interface{}, policyChangeType Policy
 		if err != nil {
 			return err
 		}
-		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*inputPolicy.Spec.Background)
+		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(inputPolicy.Spec.Background)
 		policyType := metrics.Namespaced
 		policyNamespace := inputPolicy.ObjectMeta.Namespace
 		policyName := inputPolicy.ObjectMeta.Name

--- a/pkg/metrics/policyruleexecutionlatency/policyRuleExecutionLatency.go
+++ b/pkg/metrics/policyruleexecutionlatency/policyRuleExecutionLatency.go
@@ -64,7 +64,7 @@ func (pm PromMetrics) ProcessEngineResponse(policy kyverno.ClusterPolicy, engine
 		return err
 	}
 	policyType := metrics.Namespaced
-	policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*policy.Spec.Background)
+	policyBackgroundMode := metrics.ParsePolicyBackgroundMode(policy.Spec.Background)
 	policyNamespace := policy.ObjectMeta.Namespace
 	if policyNamespace == "" {
 		policyNamespace = "-"

--- a/pkg/metrics/policyruleinfo/policyRuleInfo.go
+++ b/pkg/metrics/policyruleinfo/policyRuleInfo.go
@@ -50,7 +50,7 @@ func (pm PromMetrics) AddPolicy(policy interface{}) error {
 		if err != nil {
 			return err
 		}
-		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*inputPolicy.Spec.Background)
+		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(inputPolicy.Spec.Background)
 		policyType := metrics.Cluster
 		policyNamespace := "" // doesn't matter for cluster policy
 		policyName := inputPolicy.ObjectMeta.Name
@@ -69,7 +69,7 @@ func (pm PromMetrics) AddPolicy(policy interface{}) error {
 		if err != nil {
 			return err
 		}
-		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*inputPolicy.Spec.Background)
+		policyBackgroundMode := metrics.ParsePolicyBackgroundMode(inputPolicy.Spec.Background)
 		policyType := metrics.Namespaced
 		policyNamespace := inputPolicy.ObjectMeta.Namespace
 		policyName := inputPolicy.ObjectMeta.Name
@@ -96,7 +96,7 @@ func (pm PromMetrics) RemovePolicy(policy interface{}) error {
 			if err != nil {
 				return err
 			}
-			policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*inputPolicy.Spec.Background)
+			policyBackgroundMode := metrics.ParsePolicyBackgroundMode(inputPolicy.Spec.Background)
 			policyType := metrics.Cluster
 			policyNamespace := "" // doesn't matter for cluster policy
 			policyName := inputPolicy.ObjectMeta.Name
@@ -114,7 +114,7 @@ func (pm PromMetrics) RemovePolicy(policy interface{}) error {
 			if err != nil {
 				return err
 			}
-			policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*inputPolicy.Spec.Background)
+			policyBackgroundMode := metrics.ParsePolicyBackgroundMode(inputPolicy.Spec.Background)
 			policyType := metrics.Namespaced
 			policyNamespace := inputPolicy.ObjectMeta.Namespace
 			policyName := inputPolicy.ObjectMeta.Name

--- a/pkg/metrics/policyruleresults/policyRuleResults.go
+++ b/pkg/metrics/policyruleresults/policyRuleResults.go
@@ -2,11 +2,12 @@ package policyruleresults
 
 import (
 	"fmt"
+	"time"
+
 	kyverno "github.com/kyverno/kyverno/pkg/api/kyverno/v1"
 	"github.com/kyverno/kyverno/pkg/engine/response"
 	"github.com/kyverno/kyverno/pkg/metrics"
 	prom "github.com/prometheus/client_golang/prometheus"
-	"time"
 )
 
 func (pm PromMetrics) registerPolicyRuleResultsMetric(
@@ -57,7 +58,7 @@ func (pm PromMetrics) ProcessEngineResponse(policy kyverno.ClusterPolicy, engine
 		return err
 	}
 	policyType := metrics.Namespaced
-	policyBackgroundMode := metrics.ParsePolicyBackgroundMode(*policy.Spec.Background)
+	policyBackgroundMode := metrics.ParsePolicyBackgroundMode(policy.Spec.Background)
 	policyNamespace := policy.ObjectMeta.Namespace
 	if policyNamespace == "" {
 		policyNamespace = "-"

